### PR TITLE
feat: Added Navigation Component

### PR DESCRIPTION
**Pull request by mnath30**
Added navigation components:
1. Default Navigation
2. Navigation with only menu items
3. E-commerce navigation example
4. Social Media navigation example

https://user-images.githubusercontent.com/55870039/152955865-3b3fc886-ec07-4e2e-81d0-85a674771dd6.mp4

### CODE REVIEW

**Review by Bhakti199 — APPROVED** [review]
code is nicely written and structured, good to merge

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (dev-enforced) [inline .vscode/settings.json]: You can add this file to .gitignore as it is not correct to add Ide settings over here.

**Review by dev-enforced — APPROVED** [review]
Code looks good to me. Well structured documentation. You can merge it @mnath30 .Great work!!

File: .vscode/settings.json (hunk 1 above)
Comment (mnath30) [reply .vscode/settings.json]: Done. Added gitignore.

**Pull request merged**